core/quantity/src/Formatter/Format.ts

```diff
@@ -9,319 +9,200 @@
-      case 16:
-        return FractionalPrecision.Sixteen;
-      case 32:
-        return FractionalPrecision.ThirtyTwo;
-      case 64:
-        return FractionalPrecision.SixtyFour;
-      case 128:
-        return FractionalPrecision.OneHundredTwentyEight;
-      case 256:
-        return FractionalPrecision.TwoHundredFiftySix;
-      default:
-        throw new QuantityError(QuantityStatus.InvalidJson, `The Format ${formatName} has an invalid 'precision' attribute.`);
+    if (undefined !== formatProps.uomSeparator) { // optional; default is " "
+      if (typeof (formatProps.uomSeparator) !== "string")
+        throw new QuantityError(QuantityStatus.InvalidJson, `The Format ${this.name} has an invalid 'uomSeparator' attribute. It should be of type 'string'.`);
+      if (formatProps.uomSeparator.length < 0 || formatProps.uomSeparator.length > 1)
+        throw new QuantityError(QuantityStatus.InvalidJson, `The Format ${this.name} has an invalid 'uomSeparator' attribute. It should be an empty or one character string.`);
+      this._uomSeparator = formatProps.uomSeparator;
     }
-  }
 
-  /** This method validates the input value, that is typically extracted for persisted JSON data, is a valid DecimalPrecision or FractionalPrecision. Throws exception if not valid. */
-  public static parsePrecision(precision: number, formatName: string, type: FormatType): DecimalPrecision | FractionalPrecision {
-    switch (type) { // type must be decimal, fractional, scientific, or station
-      case FormatType.Decimal:
-      case FormatType.Scientific:
-      case FormatType.Station:
-        return Format.parseDecimalPrecision(precision);
-      case FormatType.Fractional:
-        return Format.parseFractionalPrecision(precision, formatName);
+    if (undefined !== formatProps.stationSeparator) { // optional; default is "+"
+      if (typeof (formatProps.stationSeparator) !== "string")
+        throw new QuantityError(QuantityStatus.InvalidJson, `The Format ${this.name} has an invalid 'stationSeparator' attribute. It should be of type 'string'.`);
+      if (formatProps.stationSeparator.length > 1)
+        throw new QuantityError(QuantityStatus.InvalidJson, `The Format ${this.name} has an invalid 'stationSeparator' attribute. It should be an empty or one character string.`);
+      this._stationSeparator = formatProps.stationSeparator;
     }
   }
+}
 
-  private verifyFormatTraitsOptions(formatTraitsFromJson: string | string[]) {
-    const formatTraits = (Array.isArray(formatTraitsFromJson)) ? formatTraitsFromJson : formatTraitsFromJson.split(/,|;|\|/);
-    formatTraits.forEach((formatTraitsString: string) => { // for each element in the string array
-      this._formatTraits = Format.parseFormatTrait(formatTraitsString, this.formatTraits);
-    });
+/** A class used to define the specifications for formatting quantity values. This class is typically loaded by reading [[FormatProps]].
+ * @beta
+ */
+export class Format extends BaseFormat {
```

Comment:
Why can't the Format class in ecschema-metadata directly use this format class rather having to create a base class to work with? I know it would have to wrap this one but not sure I'm seeing why we need two.

Reply:
I know we discussed this with Colin, so I will go ahead and resolve this conversation, if that's OK.